File: Chapter16/SampleViewAnimation/Java/SampleViewAnimation/app/src/main/java/com/ai/android/ExerciseSystemIntents/ViewAnimationListener.java
package com.ai.android.ExerciseSystemIntents;

import android.util.Log;
import android.view.animation.Animation;

public class ViewAnimationListener implements Animation.AnimationListener {

    @Override
    public void onAnimationStart(Animation animation) {
        Log.d("Animation Example", "onAnimationStart");
    }

    @Override
    public void onAnimationEnd(Animation animation) {
        Log.d("Animation Example", "onAnimationEnd");
    }

    @Override
    public void onAnimationRepeat(Animation animation) {
        Log.d("Animation Example", "onAnimationRepeat");
    }
}

File: Chapter16/SampleViewAnimation/Kotlin/SampleViewAnimation/app/src/main/java/com/ai/android/sampleviewanimation/ViewAnimationListener.kt
package com.ai.android.sampleviewanimation

import android.util.Log
import android.view.animation.Animation

class ViewAnimationListener: Animation.AnimationListener {

    override fun onAnimationStart(animation: Animation?) {
        Log.d(TAG, "onAnimationStart")
    }

    override fun onAnimationEnd(animation: Animation?) {
        Log.d(TAG, "onAnimationEnd")
    }

    override fun onAnimationRepeat(animation: Animation?) {
        Log.d(TAG, "onAnimationRepeat")
    }
    
    companion object {
        const val TAG = "ViewAnimationListener"
    }
}